kernel/src/net/socket/ip/stream/init.rs

```diff
@@ -90,19 +147,66 @@ impl InitStream {
 
         match ListenStream::new(bound_port, backlog, option, observer) {
             Ok(listen_stream) => Ok(listen_stream),
-            Err((bound_port, error)) => Err((error, Self::Bound(bound_port))),
+            Err((bound_port, error)) => Err((error, Self::new_bound(bound_port))),
         }
     }
 
-    pub fn local_endpoint(&self) -> Option<IpEndpoint> {
-        match self {
-            InitStream::Unbound => None,
-            InitStream::Bound(bound_port) => Some(bound_port.endpoint().unwrap()),
+    pub fn try_recv(
+        &self,
+        _writer: &mut dyn MultiWrite,
+        _flags: SendRecvFlags,
+    ) -> Result<(usize, SocketAddr)> {
```

Comment:
Since `InitStream` cannot actually send and receive, there's no need to pass `writer` and `flags` as parameters?